ledger/nomos-ledger/src/mantle/sdp/rewards.rs

```diff
@@ -0,0 +1,78 @@
+use std::collections::HashMap;
+
+use nomos_core::{
+    block::BlockNumber,
+    sdp::{ActivityMetadata, DeclarationId, ProviderId, ServiceParameters},
+};
+
+use super::SessionState;
+
+pub type RewardAmount = u64;
+
+/// Generic trait for service-specific reward calculation.
+///
+/// Each service can implement its own rewards logic by implementing this trait.
+/// The rewards object is updated with active messages and session transitions,
+/// and can calculate expected rewards for each provider based on the service's
+/// internal logic.
+pub trait Rewards: Clone + PartialEq + Eq + Send + Sync + std::fmt::Debug {
+    /// Update rewards state when an active message is received.
+    ///
+    /// Called when a provider submits an active message with optional metadata
```

Comment:
metadata is no longer optional IIRC
```suggestion
    /// Called when a provider submits an active message with metadata
```